hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/StreamReadMonitoringFunction.java

```diff
@@ -207,34 +234,46 @@ private HoodieTableMetaClient getOrCreateMetaClient() {
   }
 
   @VisibleForTesting
-  public void monitorDirAndForwardSplits(SourceFunctionAdapter.SourceContext<MergeOnReadInputSplit> context) {
+  public void monitorDirAndForwardSplits(SourceContext<MergeOnReadInputSplit> context) {
     HoodieTableMetaClient metaClient = getOrCreateMetaClient();
     if (metaClient == null) {
       // table does not exist
       return;
     }
-    IncrementalInputSplits.Result result =
-        incrementalInputSplits.inputSplits(metaClient, this.issuedOffset, this.cdcEnabled);
+
+    IncrementalInputSplits.Result result = remainingSplits.isEmpty()
+        ? incrementalInputSplits.inputSplits(metaClient, this.issuedOffset, this.cdcEnabled)
+        : IncrementalInputSplits.Result.instance(remainingSplits, issuedInstant, issuedOffset);
 
     if (result.isEmpty() && StringUtils.isNullOrEmpty(result.getEndInstant())) {
       // no new instants, returns early
       LOG.warn("No new instants to read for current run.");
       return;
     }
 
-    for (MergeOnReadInputSplit split : result.getInputSplits()) {
-      context.collect(split);
+    List<MergeOnReadInputSplit> inputSplits = result.getInputSplits();
+    LOG.info("Table {} : Read {} inputsplits for current instant", conf.get(FlinkOptions.TABLE_NAME), inputSplits.size());
+    int endIndex = Math.min(splitsLimit, inputSplits.size());
+    for (int index = 0; index < endIndex; index++) {
+      context.collect(inputSplits.get(index));
+    }
+    remainingSplits = inputSplits.stream().skip(endIndex).collect(Collectors.toList());
+    if (totalSplits < 0 || !result.getEndInstant().equals(issuedInstant)) {
+      totalSplits = inputSplits.size();
     }
 
     // update the issues instant time
     this.issuedInstant = result.getEndInstant();
     this.issuedOffset = result.getOffset();
+    int sentSplits = totalSplits - remainingSplits.size();
+    double sentPercent = totalSplits == 0 ? 0 : (1 - remainingSplits.size() / (double) totalSplits) * 100;
     LOG.info("\n"
             + "------------------------------------------------------------\n"
             + "---------- table: {}\n"
             + "---------- consumed to instant: {}\n"
+            + "---------- total sent {} inputsplits out of {}, percent is {}%\n"
```

Comment:
`total sent {} inputsplits out of {}, percent is` -> `sent {} inputsplits out of {}, percentage is`